Remove listGroups while building categories

MrFlashAccount opened the pull request:
### Pull Request Description

This PR adds new field called `groups` on the User object to avoid fetching the usergroups list to get the names of the groups. 

Also this PR removes the `<ManagePermisssionsModal />` 

<!--
- Please describe the nature of your PR here, as well as the motivation for it.
- If it fixes an open issue, please mention that issue number here.
-->

### Important Notes

<!--
- Mention important elements of the design.
- Mention any notable changes to APIs.
-->

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.
- [ ] If meaningful changes were made to logic or tests affecting Enso Cloud integration in the libraries, 
      or the Snowflake database integration, a run of the [Extra Tests](https://github.com/enso-org/enso/actions/workflows/extra-nightly-tests.yml) has been scheduled.
  - If applicable, it is suggested to paste a link to a successful run of the Extra Tests.

MrFlashAccount committed:
Remove listGroups while building categories

somebody1234 reviewed (approved):
CR :white_check_mark:

MrFlashAccount committed:
Small adjustment

MrFlashAccount committed:
Fix lint errors

MrFlashAccount committed:
Fix types

MrFlashAccount commented:
🚫 QA:

1. Moving assets in and out of the assets tree breaks the dashboard: 
   
https://github.com/user-attachments/assets/eba53e62-a886-4462-8205-a8aa9807dd03

2. Perf regression: Refetching re-renders the whole Table
   
https://github.com/user-attachments/assets/13a1c8d8-4639-472c-a48b-840c0b39d8e6

3. I think for the assets table it's better to invalidate both active and inactive queries.

MrFlashAccount committed:
Fix types

MrFlashAccount committed:
Some fixes in types

MrFlashAccount committed:
Merge branch 'develop' into wip/sergeigarin/remove-list-usergroups

# Conflicts:
#	app/gui/src/dashboard/hooks/backendHooks.tsx
#	app/gui/src/dashboard/layouts/AssetsTable.tsx

hubertp committed:
Better detection for Enso's NI when launching Language Server (#12034)

Previosly we were wrongly relying on the presence of the file. That way,
a bash script meant that NI integration was wrongly used.
This change uses Tika, but it has already been present in other
subprojects so no additional dependency shall be added.

Follow up on #11880.

MrFlashAccount committed:
Remove manage permissions modal

MrFlashAccount committed:
FIx types

MrFlashAccount committed:
Fix types

MrFlashAccount committed:
Fix types -_-

MrFlashAccount committed:
Remove unused var

MrFlashAccount committed:
Refactor Setup Organization dialog

MrFlashAccount commented:
Re-requesting CR as this PR changed after the latest code review

MrFlashAccount committed:
Fix eslint

MrFlashAccount committed:
Fix lint

MrFlashAccount committed:
Include IDE changes into GUI workflow

MrFlashAccount committed:
Fix lint

somebody1234 commented on build.sbt:
```diff
@@ -4495,10 +4495,12 @@ lazy val `runtime-version-manager` = project
     libraryDependencies ++= Seq(
       "com.typesafe.scala-logging" %% "scala-logging"    % scalaLoggingVersion,
       "org.apache.commons"          % "commons-compress" % commonsCompressVersion,
+      "org.apache.tika"             % "tika-core"        % tikaVersion,
```
why is this part of the changeset?

MrFlashAccount replied:
Shouldn't be. I think it's a bad merge or something. Will fix, thank you!

somebody1234 reviewed (approved):
CR :white_check_mark:
would like to get confirmation that we want to remove the manage permissions modal though

JaroslavTulach commented on distribution/project-manager/THIRD-PARTY/org.apache.tika.tika-core-2.4.1/NOTICE:
```diff
@@ -0,0 +1,8 @@
+
```
No changes to `project-manager` in this PR please.

MrFlashAccount replied:
Yeah, this was a bad merge or something

MrFlashAccount replied:
@JaroslavTulach Please unblock the PR 🙏

JaroslavTulach reviewed (dismissed):
No changes to `project-manager`, its `legal`, `lib/scala`, please.

MrFlashAccount committed:
Merge branch 'develop' into wip/sergeigarin/remove-list-usergroups

hubertp committed:
Revert "Disable visualizations for subexpressions (#11949)" (#12050)

This reverts commit 1f763f7800c747a734074687a74671ac921a5d1a.

MrFlashAccount committed:
Invalidate queries when the user goes online + Fix offline mode (#11944)

Closes: https://github.com/enso-org/cloud-v2/issues/1630

1. Invalidate remote backend queries when user goes online
2. Moves auth-related logic from `AuthProvider` to `SessionProvider`
3. Do not log user out if `users/me` returns 401

PabloBuchu commented:
1. If I switch between accounts (Team -> Free -> Team) I am getting set orgzanization name modal desptie that my org has name. 
https://github.com/user-attachments/assets/4e3fa884-0cc4-42da-b117-19ad29f7261b

MrFlashAccount committed:
Fix Login/logout cache invalidation

MrFlashAccount commented:
@PabloBuchu should be addressed now

MrFlashAccount committed:
Merge branch 'develop' into wip/sergeigarin/remove-list-usergroups

# Conflicts:
#	app/gui/src/dashboard/layouts/UserMenu.tsx
#	app/gui/src/dashboard/providers/SessionProvider.tsx

MrFlashAccount committed:
Fix imports

MrFlashAccount committed:
Fix ci

MrFlashAccount committed:
Lint x2

MrFlashAccount committed:
Small fixes

MrFlashAccount committed:
Try fix activity

MrFlashAccount committed:
Fix tests

MrFlashAccount committed:
fix

MrFlashAccount committed:
Fix login

MrFlashAccount committed:
FIx tests x2

MrFlashAccount committed:
Fix

MrFlashAccount committed:
Fix login

MrFlashAccount committed:
Fix login
